sentence_transformers/util.py

```diff
@@ -419,86 +419,43 @@ def community_detection(embeddings, threshold=0.75, min_community_size=10, batch
 ######################
 
 
+def is_sentence_transformer_model(model_name_or_path: str, token: Optional[Union[bool, str]] = None) -> bool:
+    if os.path.exists(model_name_or_path):
+        return os.path.exists(os.path.join(model_name_or_path, "modules.json"))
```

Comment:
not sure if `modules.json` is deterministic, maybe `sentence_bert_config` is better? https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2/blob/main/sentence_bert_config.json

Reply:
I think `modules.json` must always be there, while `sentence_bert_config.json` is only *usually* there. For example, this code makes me believe that there may be models that don't use `sentence_bert_config.json`:
https://github.com/UKPLab/sentence-transformers/blob/c5f93f70eca933c78695c5bc686ceda59651ae3b/sentence_transformers/models/Transformer.py#L137-L141

And I can find some older models that don't even use Transformers that do still have `modules.json`: https://huggingface.co/sentence-transformers/average_word_embeddings_levy_dependency/tree/main

I appreciate you looking into this though!

Reply:
i see it make sense. To be frankly the way it current handle config is not so beautiful lol, maybe later a better design is needed :)

Reply:
I agree completely! I've already started brainstorming some options, though I think my main idea would be infeasible due to some breaking changes here and there. It involves subclassing `SentenceTransformer` as `PreTrainedModel` rather than `nn.Sequential`. Then it could use more of the functionality from `transformers`, e.g. `load_in_8bit`, PEFT, etc.

The `modules.json` and `sentence_bert_config.json` would be removed in favor of placing that information inside of `config_sentence_transformers.json`, and the other folders with e.g. `..._Pooling`, `..._Dense` or `..._Normalize` could be removed as well. The configuration for that would live inside of the 1 config file: `config_sentence_transformers.json`, and the weights (e.g. for Dense) would be stored natively by `transformers` via `save_pretrained` because the SentenceTransformer is now a special subclass of `PreTrainedModel`.

My primary concerns are models that don't use `transformers`, but those are few and far between.

Reply:
i'm not familiar as you on the transofmrers source code, let me read a bit the `class:PretrainedModel` and get back to you with some proper thoughts